Refactored code structure and improved UI responsiveness.

AbdulrahmanAhmeed opened the pull request.

AbdulrahmanAhmeed committed:
Refactored code structure and improved UI responsiveness.

AbdulrahmenCanadaHitech committed:
Refactored CameraStreamViewModel to use private fields with camelCase naming convention and added MotionThreshold property for motion detection configuration.

elmerbool commented on IPCameraViewer/Models/CameraStreamViewModel.cs:
```diff
@@ -0,0 +1,103 @@
+        public int Id
+        {
+            get => this.id;
+            set => SetProperty(ref this.id, value);
```
Prefix with "this."

elmerbool commented on IPCameraViewer/Models/CameraStreamViewModel.cs:
```diff
@@ -0,0 +1,103 @@
+
+        protected virtual void OnPropertyChanged([CallerMemberName] string? propertyName = null)
+        {
+            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
```
Prefix with "this."

elmerbool commented on IPCameraViewer/Models/CameraStreamViewModel.cs:
```diff
@@ -0,0 +1,103 @@
+                return false;
+
+            field = value;
+            OnPropertyChanged(propertyName);
```
Prefix with "this."

elmerbool commented on IPCameraViewer/Models/CameraStreamViewModel.cs:
```diff
@@ -0,0 +1,103 @@
+
+        protected bool SetProperty<T>(ref T field, T value, [CallerMemberName] string? propertyName = null)
+        {
+            if (EqualityComparer<T>.Default.Equals(field, value))
```
Use single "return" statement for whole method.

elmerbool commented on IPCameraViewer/Services/MjpegStreamer.cs:
```diff
@@ -0,0 +1,204 @@
+							var frameBytes = readBuffer.GetRange(soi, length).ToArray();
+							readBuffer.RemoveRange(0, eoi + 2);
+
+							OnFrame(frameBytes);
```
Prefix with ".this"

elmerbool commented on IPCameraViewer/Services/MjpegStreamer.cs:
```diff
@@ -0,0 +1,204 @@
+
+						while (true)
+						{
+							int soi = IndexOfPattern(readBuffer, 0xFF, 0xD8);
```
Prefix with ".this"

elmerbool commented on IPCameraViewer/Services/MjpegStreamer.cs:
```diff
@@ -0,0 +1,204 @@
+			}
+			catch (Exception ex)
+			{
+				Error?.Invoke(ex.Message);
```
Prefix with ".this"

elmerbool commented on IPCameraViewer/Services/MjpegStreamer.cs:
```diff
@@ -0,0 +1,204 @@
+
+		private void OnFrame(byte[] jpegBytes)
+		{
+			FrameReceived?.Invoke(jpegBytes);
```
Prefix with "this."

elmerbool commented on IPCameraViewer/Services/MjpegStreamer.cs:
```diff
@@ -0,0 +1,204 @@
+
+		private static int IndexOfPattern(List<byte> data, byte b1, byte b2, int start = 0)
+		{
+			for (int i = start; i < data.Count - 1; i++)
```
Use single "return" statement only.

elmerbool commented on IPCameraViewer/Services/MjpegStreamer.cs:
```diff
@@ -0,0 +1,204 @@
+
+				if (fireMotion)
+				{
+					MotionDetected?.Invoke();
```
Prefix with "this."

elmerbool commented on IPCameraViewer/Services/MjpegStreamer.cs:
```diff
@@ -0,0 +1,204 @@
+			}
+			catch (Exception ex)
+			{
+				Error?.Invoke(ex.Message);
```
Prefix with "this."

elmerbool commented on IPCameraViewer/MainPage.xaml.cs:
```diff
@@ -0,0 +1,232 @@
+            {
+                StopStream(stream);
+            }
+            UpdateStatus("All streams stopped");
```
Prefix with "this."

elmerbool commented on IPCameraViewer/MainPage.xaml.cs:
```diff
@@ -0,0 +1,232 @@
+            {
+                stream.DetectionLogs.Clear();
+            }
+            UpdateStatus("All logs cleared");
```
Prefix with "this."

elmerbool commented on IPCameraViewer/MainPage.xaml.cs:
```diff
@@ -0,0 +1,232 @@
+        {
+            if (streamViewModel.Streamer != null)
+            {
+                StopStream(streamViewModel);
```
Prefix with "this."

elmerbool commented on IPCameraViewer/MainPage.xaml.cs:
```diff
@@ -0,0 +1,232 @@
+            base.OnDisappearing();
+            foreach (var stream in this.streams)
+            {
+                StopStream(stream);
```
Prefix with "this."

elmerbool commented on IPCameraViewer/MainPage.xaml.cs:
```diff
@@ -0,0 +1,232 @@
+            CameraUrlEntry.Text = string.Empty;
+            CameraNameEntry.Text = string.Empty;
+
+            UpdateStatus($"Added stream: {cameraName}");
```
Prefix with "this."

elmerbool commented on IPCameraViewer/MainPage.xaml.cs:
```diff
@@ -0,0 +1,232 @@
+            if (string.IsNullOrWhiteSpace(cameraUrl))
+            {
+                DisplayAlert("Error", "Please enter a valid camera URL.", "OK");
+                return;
```
Use single "return" statement.

AbdulrahmanAhmeed committed:
Add Motion Threshold Slider and enhance stream management

- Introduced a slider for adjusting the motion threshold percentage in the UI.
- Updated the CameraStreamViewModel to use MotionThresholdPercent for motion detection configuration.
- Improved input validation in the stream addition process.
- Enhanced the MjpegStreamer to utilize the new motion threshold ratio for motion detection.
- Refactored code for better readability and consistency.

AbdulrahmanAhmeed committed:
Refactor MainPage.xaml.cs for improved readability and consistency

- Introduced constants for various alert messages and status formats to enhance maintainability.
- Updated string formatting for status updates and alerts to use the new constants.
- Refactored stream management methods to improve clarity and consistency in method calls.
- Enhanced the BoolToStartStopConverter for better readability.
- Made minor adjustments in CameraStreamViewModel for consistency in property access.

AbdulrahmanAhmeed committed:
Refactor CameraStreamViewModel and MjpegStreamer for consistency and clarity

- Updated property accessors in CameraStreamViewModel to use 'this.SetProperty' for consistency.
- Refactored method calls in MjpegStreamer to use the class name for static method access, enhancing code clarity.
